src/main/java/io/github/syst3ms/skriptparser/registration/ContextValueManager.java

```diff
@@ -0,0 +1,111 @@
+package io.github.syst3ms.skriptparser.registration;
+
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class ContextValueManager {
```

Comment:
I think these classes should be restructured and renamed. First, I would rename ContextValueManager to ContextValues, and ContextValue to ContextValueInfo. 
On top of these, the constants in ContextValue (which should be final by the way) should indeed be moved to a separate enum, named something like ContextValueTime.

All these classes would go in a separate package as well.

Reply:
I especially didn’t name it ContextValueInfo since it clashes with the naming conventions in my opinion. It doesn’t implement SyntaxInfo (a bit like Converters).

Should the package be a subpackage of the current registration package? I think that’s what you meant since a whole package for this seems a bit over the top.

Reply:
I suppose ContextValue can stay as such then. And yes, I meant a subpackage.